protos/logging/logging.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package mavsdk.rpc.logging;
+
+option java_package = "io.mavsdk.logging";
+option java_outer_classname = "LoggingProto";
+
+service LoggingService {
+    // Send command to start streaming logging data over MAVLink
+    rpc StartLogging(StartLoggingRequest) returns(StartLoggingResponse) {}
+    // Send command to stop streaming logging data over MAVLink
+    rpc StopLogging(StopLoggingRequest) returns(StopLoggingResponse) {}
+    // Subscribe to logging messages
+    rpc SubscribeLoggingRaw(SubscribeLoggingRawRequest) returns(stream LoggingRaw) {}
+}
+
+message StartLoggingRequest {}
+message StartLoggingResponse {
+    LoggingResult logging_result = 1;
+}
+
+message StopLoggingRequest {}
+message StopLoggingResponse {
+    LoggingResult logging_result = 1;
+}
+
+message SubscribeLoggingRawRequest {}
+message SubscribeLoggingRawResponse {
+    LoggingRaw logging_raw = 1; // A message containing logged data
+}
+
+message LoggingRaw {
+    uint32 first_message_offset = 1; // Offset into data where first message starts
```

Comment:
So how exactly would this work? Why does it require this offset?

I'm asking because my opinion is that the API of MAVSDK should be as easy as possible, so I wonder how this would be used.

Reply:
This is internal method to recover offset if data was lost
(https://mavlink.io/en/messages/common.html#LOGGING_DATA).
This will helpful for OTA processing, for example.

Reply:
Right, so this essentially exposes the actual MAVLink protocol? However, it internally would take care of retransmission of missed messages, right?

Reply:
Retransmission is not available now in PX4 because of logging MAVLink stream require high bandwidth and CPU resources.

So we haven't way to request lost messages internal in MAVSDK.

I see only one way to avoid to send non-consistent data - parse [ULOG messages](https://dev.px4.io/v1.9.0/en/log/ulog_file_format.html) while receiving LOGGING_DATA MAVLink messages.
Without it backend should transmit full message's payload, I think.

Reply:
Aha. What about the headers, etc.? Are they just part of the first messages?

Reply:
Yes, it is received in first 16 bytes on streaming start (I'm already checked that).

Reply:
ok

Reply:
@julianoes now Im looking at QGC implementation of [logging stream processing](https://github.com/mavlink/qgroundcontrol/blob/2f5324fd1745aa7ffa8ab1d477e2c8f1cad61c1c/src/Vehicle/MAVLinkLogManager.cc#L238) and I see that may exist at least two ways:

- Transieve in gRPC `first_message_offset`, `data` and `seguence` to full processing on the other side of gRPC.
- Create 'message LoggingMessage` and `message LoggingFileHeader' and implement some processing like QGC in MAVSDK.

I think that both ways have an advantages and disadvantages.

Which case seems like preferred for you? Or may be you have any other idea?

Reply:
Or we can pasre `msg_id` in header and stream every messages as a custom gRPC `messages`

Reply:
> Create message LoggingFileHeader and message LoggingMessage (it will contain only message raw data) and implement some processing like QGC in MAVSDK

I think that makes sense.

Reply:
I would like to abstract it away as much as possible to have the easiest API.